fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/sorting/MemoryDam.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * MemoryDam.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2021 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.apple.foundationdb.record.sorting;
+
+import com.apple.foundationdb.annotation.API;
+import com.apple.foundationdb.record.provider.common.StoreTimer;
+import com.google.common.base.Suppliers;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.AbstractCollection;
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.LinkedHashMap;
+import java.util.LinkedHashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeMap;
+import java.util.function.Supplier;
+
+/**
+ * Collect keyed values into a {@link TreeMap} so that they end up sorted.
```

Comment:
I assume this is copied and pasted incorrectly, given that this uses a `LinkedHashMap` internally, not a `TreeMap`

Reply:
I believe this is still out of date

Reply:
done